app/scripts/lib/rpc-method-middleware/createUnsupportedMethodMiddleware.ts

```diff
@@ -1,18 +1,18 @@
 import type { JsonRpcMiddleware } from '@metamask/json-rpc-engine';
 import type { JsonRpcParams } from '@metamask/utils';
 import { rpcErrors } from '@metamask/rpc-errors';
-import { UNSUPPORTED_RPC_METHODS } from '../../../../shared/constants/network';
 
 /**
  * Creates a middleware that rejects explicitly unsupported RPC methods with the
  * appropriate error.
+ *
+ * @param methods - The list of unsupported RPC methods.
  */
-export function createUnsupportedMethodMiddleware(): JsonRpcMiddleware<
-  JsonRpcParams,
-  null
-> {
+export function createUnsupportedMethodMiddleware(
+  methods: string[],
```

Comment:
Should this default to `UNSUPPORTED_RPC_METHODS`?

Reply:
I had that as the default value on this line before but pulled it out. I think it makes more sense to pull it out? or do you feel that it shouldn't be the concern of the caller/instantiater of this middleware in default/EIP-1193 case?

Reply:
I would prefer having a default, but not blocking on it

Reply:
Default added here https://github.com/MetaMask/metamask-extension/pull/27782/commits/75db2f7137e3946815b26c2508d02b0542075b8a